python-frontend/src/main/java/org/sonar/python/tree/NameImpl.java

```diff
@@ -91,4 +93,9 @@ public void setSymbol(Symbol symbol) {
   public void setUsage(Usage usage) {
     this.usage = usage;
   }
+
+  @Override
+  public InferredType type() {
+    return ((SymbolImpl) symbol).inferredType();
```

Comment:
I think that `symbol` might be null here.

Reply:
Fixed in b9f4ed6.